packages/ra-ui-materialui/src/button/BulkUpdateWithConfirmButton.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import { FC, Fragment, useState, ReactElement } from 'react';
+import PropTypes from 'prop-types';
+import ActionUpdate from '@material-ui/icons/Update';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import inflection from 'inflection';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    useTranslate,
+    useUpdateMany,
+    useRefresh,
+    useNotify,
+    useUnselectAll,
+    CRUD_UPDATE_MANY,
+    useResourceContext,
+} from 'ra-core';
+
+import Confirm from '../layout/Confirm';
+import Button, { ButtonProps } from './Button';
+import { BulkActionProps } from '../types';
+
+const useStyles = makeStyles(
+    theme => ({
+        updateButton: {
+            color: theme.palette.error.main,
+            '&:hover': {
+                backgroundColor: fade(theme.palette.error.main, 0.12),
+                // Reset on mouse devices
+                '@media (hover: none)': {
+                    backgroundColor: 'transparent',
+                },
+            },
+        },
+    }),
+    { name: 'RaBulkUpdateWithConfirmButton' }
+);
+
+const defaultIcon = <ActionUpdate />;
+
+const BulkUpdateWithConfirmButton: FC<BulkUpdateWithConfirmButtonProps> = props => {
+    const {
+        basePath,
+        classes: classesOverride,
+        confirmTitle = 'ra.message.bulk_update_title',
+        confirmContent = 'ra.message.bulk_update_content',
+        data,
+        icon = defaultIcon,
+        label,
+        onClick,
+        selectedIds,
+        ...rest
+    } = props;
+    const [isOpen, setOpen] = useState(false);
+    const classes = useStyles(props);
+    const notify = useNotify();
+    const unselectAll = useUnselectAll();
+    const refresh = useRefresh();
+    const translate = useTranslate();
+    const resource = useResourceContext(props);
+    const [updateMany, { loading }] = useUpdateMany(
+        resource,
+        selectedIds,
+        data,
+        {
+            action: CRUD_UPDATE_MANY,
+            onSuccess: () => {
+                refresh();
+                notify('ra.notification.updated', 'info', {
+                    smart_count: selectedIds.length,
+                });
+                unselectAll(resource);
+            },
+            onFailure: error => {
+                notify(
+                    typeof error === 'string'
+                        ? error
+                        : error.message || 'ra.notification.http_error',
+                    'warning',
+                    {
+                        _:
+                            typeof error === 'string'
+                                ? error
+                                : error && error.message
+                                ? error.message
+                                : undefined,
+                    }
+                );
+                setOpen(false);
+            },
```

Comment:
We should also support custom `onSuccess` and `onFailure` prop as functions to customize side effects